iceoryx_hoofs/test/moduletests/test_console_logger.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/log/platform_building_blocks/console_logger.hpp"
+
+#include "iceoryx_hoofs/log/logging.hpp"
+#include "iceoryx_platform/fcntl.hpp"
+#include "test.hpp"
+
+#include <cstdio>
+#include <iostream>
+
+namespace
+{
+using namespace ::testing;
+
+class LoggerSUT : public iox::pbb::ConsoleLogger
+{
+  public:
+    using iox::pbb::ConsoleLogger::flush;
+    using iox::pbb::ConsoleLogger::logString;
+};
+
+#if !defined(_WIN32)
+TEST(ConsoleLogger_test, TestOutput)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "67f1dac5-b425-414a-9690-268ecb06c1ee");
+    GTEST_SKIP() << "todo iox-#1345 this does not work on the CI and needs to be refactored. An alternative would be "
+                    "to run a binary which uses the logger and capture its output.";
+
+    constexpr const char* LOG_FILE_NAME{"iceoryx_console_logger_output_test.txt"};
```

Comment:
We need a `IOX_TEMP_FILE_PATH` in platform which we can use to store all the test files.

Reply:
Yes, something like that would be helpful. I think there are already some unit tests which could make use of these. Maybe we could support all the XDG specification for tile paths in our platform abstraction.

Reply:
I created an issue for that: https://github.com/eclipse-iceoryx/iceoryx/issues/1652